sdk/core/src/services/AnchoringService.ts

```diff
@@ -0,0 +1,73 @@
+import { ELEM_ANCHORED_DID_METHOD, ELEM_DID_METHOD, JOLO_DID_METHOD } from '../_defaultConfig'
+import { RegistryApiService } from '@affinidi/internal-api-clients'
+import { KeysService } from '@affinidi/common'
+
+export class AnchoringService {
+  constructor(
+    private readonly api: RegistryApiService,
+    private readonly encryptedSeed: string,
+    private readonly password: string,
+    private readonly didDocument: any,
+    private readonly nonce: number,
+  ) {}
+
+  private async anchorDidWithJoloMethod(keysService: KeysService, seedHex: string) {
+    const did = this.didDocument.id
+    const signedDidDocument = await keysService.signDidDocument(this.didDocument)
+
+    const { body: bodyDidDocument } = await this.api.putDocumentInIpfs({ document: signedDidDocument })
+    const didDocumentAddress = bodyDidDocument.hash
+
+    const {
+      body: { digestHex },
+    } = await this.api.createAnchorTransaction({ nonce: this.nonce, did, didDocumentAddress })
+
+    const transactionSignatureJson =
+      digestHex && digestHex !== '' ? await keysService.createTransactionSignature(digestHex, seedHex) : ''
+
+    const transactionPublicKey = KeysService.getAnchorTransactionPublicKey(seedHex, JOLO_DID_METHOD)
+    const ethereumPublicKeyHex = transactionPublicKey.toString('hex')
+
+    const response = await this.api.anchorDid({
+      did,
+      didDocumentAddress,
+      ethereumPublicKeyHex,
+      transactionSignatureJson,
+      nonce: this.nonce,
+    })
+
+    return { did: response?.body?.did }
+  }
+
+  private async anchorDidWithElemMethod(anchoredDidElem: boolean) {
```

Comment:
It would be great if you could split this into two separate commits. One with extracting anchoring into a separate file, and another with changing its logic.

Right now it is really hard to see how flow has actually changing, because this commit is basically "remove a bunch of lines from one file, add a bunch of other lines to another file".